cli/cmd/publicdomain/add.go

```diff
@@ -22,30 +22,41 @@ func (r *Register) Run(ctx *clicontext.CLIContext) error {
 	domainName := ctx.CLI.Args().Get(0)
 	target := ctx.CLI.Args().Get(1)
 
-	result, err := ctx.ByID(target)
-	if err != nil {
-		return err
-	}
-
 	pd := adminv1.PublicDomain{
 		Spec: adminv1.PublicDomainSpec{
 			SecretName: r.Secret,
 		},
 	}
-	switch result.Object.(type) {
-	case *riov1.Service:
-		svc := result.Object.(*riov1.Service)
-		targetApp, targetVersion := services.AppAndVersion(svc)
-		if !strings.Contains(target, "@") {
-			targetVersion = ""
+
+	/*
+		1) if target is [namespace:]name and can't be found as an object, treat it as app
+		2) if target is [namespace:]name@version, treat it as service(revision)
+		3) if target is [namespace:]name and can be found as an object, treat it as a router
+	*/
```

Comment:
Nice comments. Would be nice to add unit or integration tests to this in long term.

Reply:
We can cover that in validation tests. I think all public domain tests are validation tests.

Reply:
cc @rancher-max ^ low priority, similar to rio build cases.